nifi-docker/dockerhub/Dockerfile

```diff
@@ -16,37 +16,38 @@
 # under the License.
 #
 
-FROM openjdk:8
-MAINTAINER Apache NiFi <dev@nifi.apache.org>
+FROM openjdk:8-jre
+LABEL maintainer "Apache NiFi <dev@nifi.apache.org>"
 
 ARG UID=1000
-ARG GID=50
+ARG GID=1000
 ARG NIFI_VERSION=1.4.0
+ARG MIRROR=https://archive.apache.org/dist
 
-ENV NIFI_BASE_DIR /opt/nifi
-ENV NIFI_HOME $NIFI_BASE_DIR/nifi-$NIFI_VERSION
-ENV NIFI_BINARY_URL https://archive.apache.org/dist/nifi/$NIFI_VERSION/nifi-$NIFI_VERSION-bin.tar.gz
+ENV NIFI_BASE_DIR /opt/nifi 
+ENV NIFI_HOME=$NIFI_BASE_DIR/nifi-$NIFI_VERSION \
+    NIFI_BINARY_URL=/nifi/$NIFI_VERSION/nifi-$NIFI_VERSION-bin.tar.gz
 
 # Setup NiFi user
-RUN groupadd -g $GID nifi || groupmod -n nifi `getent group $GID | cut -d: -f1`
-RUN useradd --shell /bin/bash -u $UID -g $GID -m nifi
-RUN mkdir -p $NIFI_HOME
+RUN groupadd -g $GID nifi || groupmod -n nifi `getent group $GID | cut -d: -f1` \
+    && useradd --shell /bin/bash -u $UID -g $GID -m nifi \
+    && mkdir -p $NIFI_HOME/conf/templates \
+    && chown -R nifi:nifi $NIFI_BASE_DIR
 
-# Download, validate, and expand Apache NiFi binary.
-RUN curl -fSL $NIFI_BINARY_URL -o $NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz \
-	&& echo "$(curl $NIFI_BINARY_URL.sha256) *$NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz" | sha256sum -c - \
-	&& tar -xvzf $NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz -C $NIFI_BASE_DIR \
-	&& rm $NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz
-
-RUN chown -R nifi:nifi $NIFI_HOME
+USER nifi
 
-# Web HTTP Port
-EXPOSE 8080
+# Download, validate, and expand Apache NiFi binary.
+RUN curl -fSL $MIRROR/$NIFI_BINARY_URL -o $NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz \
+    && echo "$(curl https://archive.apache.org/dist/$NIFI_BINARY_URL.sha256) *$NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz" | sha256sum -c - \
+    && tar -xvzf $NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz -C $NIFI_BASE_DIR \
+    && rm $NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz \
+    && chown -R nifi:nifi $NIFI_HOME
```

Comment:
Shouldn't be necessary, no. Doesn't seem to be hurting anything, though. 
I think I did it because I didn't trust the previous `chown -R` to hold after the `curl`, but since using `USER` beforehand, that does make it redundant.